### PR TITLE
tsp, add test for cadl-ranch Type/Model and Type/Property

### PR DESCRIPTION
**Pull request by v-hongli1**
Add test cases for [issue#2485](https://github.com/Azure/autorest.java/issues/2485)

### CODE REVIEW

File: typespec-tests/src/main/java/com/type/model/inheritance/nesteddiscriminator/models/Salmon.java [hunk 1]
```diff
@@ -116,11 +116,11 @@ public Salmon setPartner(Fish partner) {
     @Override
     public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
         jsonWriter.writeStartObject();
-        jsonWriter.writeStringField("kind", "salmon");
         jsonWriter.writeIntField("age", getAge());
+        jsonWriter.writeStringField("kind", "salmon");
```
Comment (haolingdong-msft) [inline typespec-tests/src/main/java/com/type/model/inheritance/nesteddiscriminator/models/Salmon.java:120]: This is directly generated, or manual change?

Comment (v-hongli1) [reply typespec-tests/src/main/java/com/type/model/inheritance/nesteddiscriminator/models/Salmon.java]: This is directly generated, so removed now.

File: typespec-tests/src/test/java/com/type/property/additionalproperties/ExtendsTests.java [hunk 2]
```diff
@@ -104,4 +108,43 @@ public void testExtendsUnknownClient() {
         Assertions.assertEquals("ExtendsUnknownAdditionalProperties", properties.getName());
         Assertions.assertIterableEquals(propertyMap.entrySet(), properties.getAdditionalProperties().entrySet());
     }
+
+    @Test
+    @Disabled("Polymorphic deserialization doesn't support multiple levels of inheritance in Jackson, https://github.com/FasterXML/jackson-databind/issues/1188")
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/property/additionalproperties/ExtendsTests.java:113]: I don't think this is the reason?

If you are not sure about it, just keep it empty.

File: typespec-tests/src/test/java/com/type/model/inheritance/NestedDiscriminatorTests.java
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/model/inheritance/NestedDiscriminatorTests.java:1]: This case would be lowest priority. We haven't decided to support the "multiple levels of inheritance" case.

File: typespec-tests/src/test/java/com/type/property/additionalproperties/ExtendsTests.java (hunk 2 above)
Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/property/additionalproperties/ExtendsTests.java]: The reason for disable has been fixed in the new version.

File: typespec-tests/src/test/java/com/type/model/inheritance/NestedDiscriminatorTests.java
Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/model/inheritance/NestedDiscriminatorTests.java:1]: Test cases `putModel` and `putRecursiveModel` have been set to disabled.

File: typespec-tests/src/test/java/com/type/model/inheritance/NestedDiscriminatorTests.java [hunk 3]
```diff
@@ -39,10 +45,35 @@ void getRecursiveModel() {
     }
 
 
-    @Disabled("Polymorphic deserialization doesn't support multiple levels of inheritance in Jackson, https://github.com/FasterXML/jackson-databind/issues/1188")
     @Test
+    @Disabled("1. The item `kind` is missing in the generated json file by the method `toJson` of the class `Shark`. " +
+            "2. The output order of each item in the generated json file by the method `toJson` of the class `Salmon` is incorrect.")
```
Comment (weidongxu-microsoft) [inline typespec-tests/src/test/java/com/type/model/inheritance/NestedDiscriminatorTests.java:50]: Why order affect the test? JSON object does not care about order, isn't it?

Comment (v-hongli1) [reply typespec-tests/src/test/java/com/type/model/inheritance/NestedDiscriminatorTests.java]: Fixed in the new version.

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
https://github.com/Azure/autorest.java/issues/2500

**Comment by weidongxu-microsoft** [comment]
some of them being fixed in https://github.com/Azure/autorest.java/pull/2504

**Pull request merged**